sebs/knative/trigger.py

```diff
@@ -0,0 +1,295 @@
+import concurrent.futures
+import datetime
+import json
+import requests
+import subprocess
+import time
+from typing import Dict, List, Optional  # noqa
+
+from sebs.faas.function import ExecutionResult, Trigger
+
+
+class LibraryTrigger(Trigger):
+    """
+    Trigger implementation for invoking a Knative service using port forwarding and curl.
+
+    Attributes:
+        function_name (str): The name of the function to invoke.
+        pod_name (str): The name of the Kubernetes pod where the function is deployed.
+        namespace (str): The Kubernetes namespace where the pod is deployed (default is 'default').
+    """
+
+    def __init__(self, function_name: str, pod_name: str, namespace: str = "default"):
+        """
+        Initialize the LibraryTrigger with the function name, pod name, and namespace.
+
+        Args:
+            function_name (str): The name of the function to invoke.
+            pod_name (str): The name of the Kubernetes pod where the function is deployed.
+            namespace (str, optional): The Kubernetes namespace where the pod is deployed (default is 'default').
+        """
+        super().__init__()
+        self.function_name = function_name
+        self.pod_name = pod_name
+        self.namespace = namespace
+
+    @staticmethod
+    def trigger_type() -> "Trigger.TriggerType":
+        """
+        Return the type of trigger (LibraryTrigger).
+
+        Returns:
+            Trigger.TriggerType: The trigger type (LibraryTrigger).
+        """
+        return Trigger.TriggerType.LIBRARY
+
+    @staticmethod
+    def get_curl_command(payload: dict) -> List[str]:
```

Comment:
I am very confused now. Why does the library trigger need a cURL command?

Reply:
Also, we have pycurl - see the main implementation of an HTTP trigger.

Reply:
this implementation was according to the knative serving the current implementation is now using the func